folium/map.py

```diff
@@ -21,7 +21,26 @@
 )
 
 
-class Layer(MacroElement):
+class Evented(MacroElement):
+    """The base class for Layer and Map
+
+    Adds the `on` method for event handling capabilities.
+
+    See https://leafletjs.com/reference.html#evented for
+    more in depth documentation. Please note that we have
+    only added the `on(<Object> eventMap)` variant of this
+    method using python keyword arguments.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def on(self, **event_map):
```

Comment:
can we add a type hint for this?

```suggestion
    def on(self, **event_map: JsCode):
```